src/testing/test_runner.cc

```diff
@@ -130,6 +131,11 @@ test_runner::run_sync(std::function<future<>()> task) {
         // it.
         return;
     }
+    if (_done) {
```

Comment:
Can it happen that _done is false, but _exit_code is non-zero?
In other words -- is the above `if (_exit_code != 0)` still required?

Reply:
ahh, you're right. with my change, if `_done` is true, it implies `app.run()` has returned -- it's an early return, and the return code is either `0` or a non-zero status code. let me update the change to further simplify this implementation.